src/datepicker/reducer/bs-datepicker.effects.ts

```diff
@@ -154,6 +154,7 @@ export class BsDatepickerEffects {
       this._store.dispatch(
         this._actions.navigateTo({
           unit: {
+            day: 1,
```

Comment:
Thanks for this PR @johnsmathew !
Could something like "if selected day not in month, set it to day 1, else re-use the selected day" be possible ?
I think it would make a smoother UI.

Reply:
From what I understood from the existing code, I don't think that is necessary. What we are passing to the `navigateTo` action is the view state. The calendar is displayed in such a way that the view state date is visible. So passing `day: 1` ensures that the correct month is displayed. Infact, passing any values from 1 to 28 would be same.

The previously selected date is stored in a separate variable and is unaffected. Until now, it was using the day from selected state for the view state as well. Maybe the PR title caused confusion. I should have named it as view state.

Reply:
You are right @johnsmathew, I just tested it and I agree the behaviour is the same, expect for #4396 that is now fixed.
(I don't know if there is precedent in the project, but you could add a small comment at the end of the line, like `// dummy value so month do not overflow` or something like that).
Thanks for the fix, I look forward to see it in a release !